Source/WebInspectorUI/UserInterface/Views/TreeElement.js

```diff
@@ -493,11 +493,16 @@ WI.TreeElement = class TreeElement extends WI.Object
 
     reveal()
     {
-        var currentAncestor = this.parent;
-        while (currentAncestor && !currentAncestor.root) {
-            if (!currentAncestor.expanded)
-                currentAncestor.expand();
-            currentAncestor = currentAncestor.parent;
+        let currentElement = this;
+        while (currentElement.parent && !currentElement.parent.root) {
+            if (!currentElement.parent.expanded)
+                currentElement.parent.expand();
+
+            // Some subclasses may hide elements by default to avoid showing too many items initially, but to reveal an
+            // element we must load that element and previous sibilings as well.
+            currentElement.parent.createElementsToRevealChildElement(currentElement);
```

Comment:
instead of having this be on the `parent`, could we maybe have `WI.DOMTreeElement` override `reveal` to do this logic instead?  IMO no reason to have a protected method if we can handle this just within `WI.DOMTreeElement` instead.  FYI it looks like `showChildNode` already does something kinda like this